text/0000-vhdl-emitter.md

```diff
@@ -0,0 +1,138 @@
+
+This contrasts with how things work for backends of a software compiler. A software compiler backend is targeting a different ISA and expected to be different. In this case, the VHDL backend can be viewed as being redundant whereas software compiler backends are orthogonal.
+
+The fact that the Verilog and VHDL backends are fungible/redundant then has problems of maintainability and validation. For maintainability, adding the VHDL backend means that any features going into either backend needs to be available in the other. This puts more of a load on existing FIRRTL developers and will slow down our cadence. For validation, you should be able to tape-out using either the VHDL or Verilog path and get the same result. This means that an implementation of this RFC would require doing some kind of equivalence checking on the Verilog and VHDL output. Doing this with open source tools may not be possible.
+
+## Open Source Tooling
+
+VHDL has no open source compiler equivalent to [Verilator](https://github.com/verilator/verilator). Some open source tools do exist, notably [GHDL](https://github.com/ghdl/ghdl), but the performance is not up to snuff with Verilator or VCS. Integration with GHDL would likely be needed as part of this PR and that adds further complexity of needing to maintain additional back-ends in testers. The existing factoring of things where post-emission back-ends live in testers may be worth reconsidering.
+
+# Rationale and alternatives
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+There's no major alternative design choice for how to implement a VHDL back-end and I think there's consensus there.
+
+If we don't do this, then we have potential problems with adoption by VHDL shops and put another hurdle that potential Chisel users have to deal with.
+
+## No VHDL Backend
+
+However, the big alternative is to not do this and push everyone onto the Verilog backend. We could try to make this more palatable by adding examples that show integration of Chisel-generated VHDL with Verilog and/or emitting VHDL blackboxes that align with generated Verilog.
+
+## Stage-based Backend
+
+A potentially cleaner implementation would be to treat the VHDL backend as a separate stage. This can still be a `Transform`, but it would be wrapped up in a separate VHDL emitter `Phase` that could be run independently. This, while deviating somewhat from the current Verilog emitter family, would potentially be easier to maintain long-term and act as a demonstration for how to refactor the Verilog emitter going forward.
+
+Nonetheless, it is likely easier to get all the VHDL emitter `Transform`s integrate with the way things work now and then refactor the emitter to be a separate stage.
+
```

Comment:
It may be possible to reuse some existing back-end. Because it may be more simple to convert between the format rather than writing emitter again. (It would be also nice to see all those HCL tools compatible with each other, this is good beginning.)